train_hierarchialXC.py

```diff
@@ -62,36 +62,40 @@
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Train your model ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 all_iters = 0
-start_time = time.time()
 cur_no = 0
-ae = GenericAutoencoder(loader.get_features().shape[1], args.input_ae_factor, args.input_ae_dim)
+ae = GenericAutoencoder(loader.get_features().shape[1], 0.25, args.input_ae_dim)
+
+INP_REC_LOSS = []
 
-for x, y in iter(data_loader):
+for epoch in range(args.epochs):
+    cur_no = 0
+    for x, _ in iter(data_loader):
+        cur_no += x.size(0)
 
-    cur_no += x.size(0)
+        inp_ae_fp = ae.forward(x.float())
+        loss_inp_rec = F.mse_loss(inp_ae_fp, x.float())
 
-    inp_ae_fp = ae.forward(x.float())
-    loss_inp_rec = F.mse_loss(inp_ae_fp, x.float())
-    inp_ae_fp = np.squeeze(inp_ae_fp.detach().numpy(), 1)
-    inp_ae_fp = csr_matrix(inp_ae_fp)
+        all_iters += 1
+        if all_iters % args.interval == 0:
+            print("{} / {} :: {} / {} - INP_REC_LOSS : {}\t"
+                  .format(epoch, args.epochs, cur_no, len_loader, round(loss_inp_rec.item(), 5)))
+        INP_REC_LOSS.append(loss_inp_rec.item())
 
-    y = np.squeeze(y.numpy(), 1)
-    y = csr_matrix(y)
+x = loader.get_features().toarray()
+x = ae.forward(torch.from_numpy(x).float())
```

Comment:
Do this in batches, and stack the data. Something like this:
```
def generate_encodings(data_loader, ae):
    vals = []
    for x, _ in iter(data_loader):
        vals.append(ae(x))
    vals = torch.cat(vals, dim=0)
    return vals
```

Use this function to generate encodings for train and test.